tabbie2.git/common/components/ArchiveExport.php

```diff
@@ -0,0 +1,242 @@
+						$sideXML->addAttribute('team', self::TEAM_PREFIX . $team->id);
+
+						$sideBallotXML = $sideXML->addChild('ballot', $result->getSpeaks($k));
+
+						$sideBallotXML->addAttribute('adjudicators', $adjs);
+						$sideBallotXML->addAttribute('ignored', 'false');
+						$sideBallotXML->addAttribute('rank', $result->{$k . '_place'});
+
+						foreach (['A', 'B'] as &$speaker)
+						{
+							$speechXML = $sideXML->addChild('speech');
+							$speechXML->addAttribute('reply', 'false');
+
+							$ballotXML = $speechXML->addChild('ballot', $result->{$k . '_' . $speaker . '_speaks'});
+							$ballotXML->addAttribute('adjudicators', $adjs);
+
+							# Test for iron-people to keep the correct speaker listed
+							$irregularKey = $k . '_irregular';
+							if ($speaker == 'A' && $result->$irregularKey == models\Team::IRREGULAR_A_NOSHOW) $speaker = 'B';
+							if ($speaker == 'B' && $result->$irregularKey == models\Team::IRREGULAR_B_NOSHOW) $speaker = 'A';
+							$speechXML->addAttribute('speaker', self::SPEAKER_PREFIX . $team->{'speaker' . $speaker . '_id'});
+						}
+					}
+				}
+			}
+
+
+			$participantsXML = $xml->addChild('participants');
+			$allInstitutions = [];
+
+			foreach ($tournament->teams as $team)
+			{
+				$teamXML = $participantsXML->addChild('team');
+				$teamXML->addAttribute('id', $team->id);
+				$teamXML->addAttribute('name', $team->name);
+				$teamXML->addAttribute('swing', $team->isSwing ? 'true' : 'false');
+
+				$breaks = [self::BREAK_CAT_PREFIX . models\User::LANGUAGE_ENL];
+				if ($team->language_status == models\User::LANGUAGE_ESL && $tournament->has_esl)
+					array_push($breaks, self::BREAK_CAT_PREFIX . models\User::LANGUAGE_ESL);
+				if ($team->language_status == models\User::LANGUAGE_EFL && $tournament->has_efl)
+					array_push($breaks, self::BREAK_CAT_PREFIX . models\User::LANGUAGE_EFL);
+				if ($team->language_status == models\User::LANGUAGE_EFL && $tournament->has_esl)
+					array_push($breaks, self::BREAK_CAT_PREFIX . models\User::LANGUAGE_ESL);
+				$teamXML->addAttribute('break-eligibilities', join(' ', $breaks));
+
+				foreach (['A', 'B'] as $speaker)
+				{
+					$user = $team->{'speaker' . $speaker};
+
+					$userXML = $teamXML->addChild('speaker', $user->getName());
+					$userXML->addAttribute('id', self::SPEAKER_PREFIX . $user->id);
+
+					$institutions = ['I'. $team->society_id];
+					foreach ($user->getCurrentSocieties() as $inst)
+					{
+						array_push($institutions, self::SOCIETY_PREFIX . $inst->id);
+						array_push($allInstitutions, $inst);
+					}
+					$userXML->addAttribute('institutions', join(' ', array_unique($institutions)));
+				}
+			}
+
+			$cores = [];
+			foreach ($tournament->cAs as $ca) array_push($cores, $ca->id);
+
+			foreach ($tournament->adjudicators as $adjudicator)
+			{
+				$institutions = [];
+				foreach ($adjudicator->getSocieties() as $inst)
+				{
+					array_push($institutions, self::SOCIETY_PREFIX . $inst->id);
```

Comment:
Are adjudicators associated with only one society in a tournament? I took it as no, but if I'm wrong, is it appropriate to release all current affiliations? Same for speakers above.